src/main/kotlin/com/ajaxproject/warehouse/service/ProductServiceImpl.kt

```diff
@@ -1,73 +1,75 @@
 package com.ajaxproject.warehouse.service
 
-import com.ajaxproject.warehouse.annotation.RateLimit
 import com.ajaxproject.warehouse.dto.ProductCreateDto
 import com.ajaxproject.warehouse.dto.ProductDataDto
 import com.ajaxproject.warehouse.dto.ProductDataLiteDto
 import com.ajaxproject.warehouse.dto.ProductUpdateDto
-import com.ajaxproject.warehouse.entity.Product
+import com.ajaxproject.warehouse.entity.MongoProduct
 import com.ajaxproject.warehouse.exception.NotFoundException
-import com.ajaxproject.warehouse.repository.ProductRepository
+import com.ajaxproject.warehouse.repository.MongoProductRepository
+import org.bson.types.ObjectId
 import org.springframework.stereotype.Service
 
 @Service
-class ProductServiceImpl(val productRepository: ProductRepository) : ProductService {
-    override fun findAllProducts(): List<ProductDataLiteDto> {
-        return productRepository.findAll().map { it.mapToLiteDto() }
-    }
+class ProductServiceImpl(
+    val mongoProductRepository: MongoProductRepository
+) : ProductService {
+
+    override fun findAllProducts(): List<ProductDataLiteDto> =
+        mongoProductRepository.findAll().map { it.mapToLiteDto() }
 
-    @RateLimit
-    override fun findById(id: Int): ProductDataDto {
-        return productRepository.findById(id)
-            .orElseThrow { NotFoundException("Product with id $id not found") }
-            .mapToDataDto()
+    override fun getById(id: String): ProductDataDto {
+        val mongoProduct: MongoProduct = mongoProductRepository.findById(ObjectId(id))
+            ?: throw NotFoundException("Product with id $id not found")
+        return mongoProduct.mapToDataDto()
     }
 
     override fun createProduct(createDto: ProductCreateDto): ProductDataDto {
-        val product = productRepository.save(createDto.mapToEntity())
+        val product: MongoProduct = mongoProductRepository.createProduct(createDto.mapToEntity())
```

Comment:
It's possible to make in one line and also why do you need to specify the specific type of product?

Reply:
You can click this button to suggest changes. JFYI

![image](https://github.com/DmitryOnopriienko/warehouse/assets/116563306/d2028e2c-80dd-4332-97f7-94958270a071)

Reply:
> It's possible to make in one line and also why do you need to specify the specific type of product?

I like explicit steps. It is more obvious what is happening here